src/nock/recorder.ts

```diff
@@ -282,10 +314,125 @@ export async function canUpdateTraffic(
 
   if (
     updateTraffic &&
-    (await exists(composeRecordingPath(recordingPath, 'new')))
+    (await exists(composeRecordingPath(recordingPath, { version: 'new' })))
   ) {
     return true;
   }
 
   return false;
 }
+
+export async function decodeRecordings(
+  recordings: RecordingDefinition[]
+): Promise<RecordingDefinition[]> {
+  return Promise.all(recordings.map(decodeRecordingResponse));
+}
+
+export async function decodeRecordingResponse(
+  recording: RecordingDefinition
+): Promise<RecordingDefinition> {
+  const contentEncoding = getResponseHeaderValue(
+    'Content-Encoding',
+    recording.rawHeaders ?? []
+  );
+
+  if (contentEncoding) {
+    recording.decodedResponse = await decodeResponse(
+      recording.response,
+      contentEncoding
+    );
+  }
+
+  return recording;
```

Comment:
nitpick: I am not a fan of changing fn parameter and returning it. It makes code harder to think about and kinda weird from type point of view - RecordingDefinition shouldn't contain property `decodedResponse` until this fn si called so fn should take RecordingDefinition and return DecodedRecordingDefinition (or something like that). You don't have to change the code if it would complicate saving/loading of recordings.

Reply:
Good point, I removed modifying of `recording`. Although you're right with having different type for decoded recordings would be better, other functions after decoding would have to check for decodedResponse property.